soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -467,9 +488,14 @@ void CheckTrackerLoadGame(int32_t fileNum) {
                 areaChecksGotten[realRcObj.rcArea]++;
             }
         }
-
-        if (areaChecksGotten[realRcObj.rcArea] != 0 || RandomizerCheckObjects::AreaIsOverworld(realRcObj.rcArea)) {
-            areasSpoiled |= (1 << realRcObj.rcArea);
+    }
+    for (int i = RCAREA_KOKIRI_FOREST; i < RCAREA_INVALID; i++) {
+        if (!IsAreaSpoiled((RandomizerCheckArea)i) && (RandomizerCheckObjects::AreaIsOverworld((RandomizerCheckArea)i) || !IS_RANDO ||
```

Comment:
since this is cpp we can use `static_cast` instead of c-style casting

Reply:
I can do that, but what's the benefit? Just for my future reference.

Reply:
`static_cast` is checked at compile time and can't fail during runtime. It as has the benefit of being more easily searchable.